### DISCUSSION TITLE
Question: Use Open-telemetry in Rails with Azure Monitor (Azure App Insights)

### DISCUSSION THREAD

**Discussion by BBB-Angelo** [open-telemetry/opentelemetry-ruby-contrib#257] [Q&A]
Hi All,

Firstly, please let me know if there is a better place to post questions - I don't want to add tickets in the Issues section but couldn't see a better place. I'm posting this after many days of searching for an answer without avail.

I'm trying to see how I can use Open-telemetry in my Rails app to gather data to Azure App Insights as the Azure Apps Insight Gem has been retired. I have tried looking at the getting started guide on the README of this repo and I thought combining it with the tutorial some specific vendors have (LogicMonitor - https://www.logicmonitor.com/blog/ruby-application-manual-instrumentation-for-distributed-traces and SigNoz) and the implementation steps for Azure Monitor with Open-telemetry in .NET (https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-enable?tabs=net) would help me out.

But, unfortunately as I'm very new to the world of Rails and haven't found a tutorial I'm not able to get the set up working.

I see the community here is quite active and involved and I would like to ask the following:
1. Does anyone know any tutorials for using Open-telemetry in Rails with Azure Monitor (Azure App Insights)?
2. Is it actually feasible to connect to Azure Monitor (Azure App Insights)? - My understanding is that Open-telemetry works with pretty much all APM tools but not sure if I'm missing something
3. Has anyone in the group attempted a similar implementation?

Many thanks in advance for taking the time to read this post.

Angelo

**Comment by plantfansam** [comment]
Hi Angelo! Thanks for the question. We don't have discussions enabled for this repo (looking into why not) so I think an issue  is fine. 

I don't know of any tutorials on the subject, but I'd recommend the following course of action:

* Confirm that Azure Monitor can accept OpenTelemetry data. It seems possible: https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-overview#send-your-telemetry. I believe you'd need to rig up the [azure monitor exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/azuredataexplorerexporter) in your opentelemetry collector.
* Get your Rails app running locally and configured to use the [console span exporter](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/sdk/lib/opentelemetry/sdk/trace/export/console_span_exporter.rb). This will let you verify that your application is properly emitting tracing data
* When deploying to your staging or QA environment, use the azure monitor exporter linked above.

Hope that helps!
